x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connectors/create_connector/create_connector.tsx

```diff
@@ -57,10 +57,8 @@ export const CreateConnector: React.FC = () => {
 
   useEffect(() => {
     // TODO: separate this to ability and preference
-    if (!selectedConnector?.isNative || !selfManagePreference) {
+    if (selectedConnector && !selectedConnector.isNative && selfManagePreference === 'native') {
```

Comment:
Would this cause the preference to mistakenly change to `native` in the following scenario?

- `selectedConnector` exists
- `selectedConnector` is not native
- `selfManagePreference` is already `selfManaged`

Reply:
With this hook the goal would be to capture when the connector selected has a value and if this connector is only available as a Self-managed force the preference to become Self-managed. I've been trying out this with all the connectors and combinations and it was working nice.